util/projectRequest.ts

```diff
@@ -11,34 +12,102 @@ export async function getProjects(): Promise<Project[]> {
     per_page: 100,
   });
 
-  const filteredData = projectsResponse.filter((repo) => !repo.archived);
+  const projects = mapReposToProjects(projectsResponse);
+  return projects;
+}
+
+//search uclaacm org for good first issues, repos which contain gfis, and match them up
+export async function getGoodFirstIssueProjects(): Promise<GFIProject[]> {
+  const PaginatedOctokit = Octokit.plugin(paginateRest);
+  const octokit = new PaginatedOctokit();
+  const gfiIssuesSearchQuery = 'org:uclaacm+label:"good+first+issue"+is:open';
+  const gfiIssuesResponse = await octokit.paginate('GET /search/issues', {
+    q: gfiIssuesSearchQuery,
+  }) as GitHubIssue[];
+  const gfiReposSearchQuery = 'good-first-issues:>0+org:uclaacm';
+  const gfiReposResponse = await octokit.paginate('GET /search/repositories', {
+    q: gfiReposSearchQuery,
+  }) as GitHubRepo[];
+  const repoMap = new Map<string, GitHubRepo>();
+  //turn the gfiReposResponse into a map of url and the entire repo
+  for (const repo of gfiReposResponse){
+    repoMap.set(repo.url,repo);
+  }
+  const gfiProjects = mapIssuesToProjects(gfiIssuesResponse, repoMap);
+  return gfiProjects;
+}
+
+function mapIssuesToProjects(issues: GitHubIssue[], repoMap: Map<string, GitHubRepo>) : GFIProject[]{
+  //turn the issues into a map of projects and a list of issues associated with it
+  const mappedRepos = new Map<string, GitHubIssue[]>();
+  // console.log(issues);
+  if(issues.length ==0){
+    //console.error('No open good first issues!');
+    return [];
+  }
```

Comment:
nit? maybe something catchable with prettier (spacing is inconsistent in this file)
```suggestion
  if (issues.length == 0) return [];
```